packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx

```diff
@@ -99,15 +107,13 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
         overscan={200}
         components={{
           EmptyPlaceholder: () =>
-            isInitialLoading ? (
-              <div className='threads-wrapper'>
-                {Array(3).fill({}).map((x, i) =>
-                  <ThreadCard
-                    key={i}
-                    showSkeleton
-                    thread={{} as any}
-                  />
-                )}
+            !isLoadedRef.current ? (
```

Comment:
1. I don't like this solution as it looks like a workaround and it is not generic. My first guess was that we should handle this using `status` or `loading` that are returned from the react-query. I looked into the code and it was something weird, because it looked like every 3 minutes the whole new query is being triggered (`data` is empty, `isPreviousData` is false etc). Decided to add one line => `keepPreviousData: true,` in `fetchThreads.ts` and apparently it did the work in this case.

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/20167c52-8f48-4ca7-8851-fbb51f7f2d61)

Can you check if with this solution works as expected?

2. Also, what I noticed is that my prettier runs on your files so I guess you still have something messed up with the config.

https://github.com/hicommonwealth/commonwealth/assets/14819225/0145cb7b-454b-45ad-82d4-cf4a00ea3a9a

3. `useFetchThreadsQuery` is not typed correctly which makes accessing properties difficult 

https://github.com/hicommonwealth/commonwealth/assets/14819225/1f932c1c-bfff-4d98-b5e9-845151128584

Reply:
Thanks for the amazing feeback @masvelio.

For 1, i had the similar concerns, already left some comments in OP.

> Other Considerations
Do we want to keep displaying old threads when new ones are fetching after the cache/stale timer? if we do then I think we should come up with a temporary memoization/state pattern to handle this case everywhere

I think this can be solved by the `keepPreviousData` as you suggested.

For 2, I did use the lint hooks but not sure why it still lints files on your side. Will give it another go.

For 3. mb there, basically the `useFetchThreadsQuery` was built incrementally so later when I added the active route in there, forgot to update the type system. I will do that in a followup if it works?

Reply:
I dont see disadvantages of using `keepPreviousData`. On the opposite, [the docs state](https://tanstack.com/query/v4/docs/react/guides/paginated-queries#better-paginated-queries-with-keeppreviousdata) that this is a "better way" of handling pagination.